sentry-ruby/lib/sentry/rack/capture_exception.rb

```diff
@@ -0,0 +1,51 @@
+module Sentry
+  module Rack
+    class CaptureException
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        # this call clones the main (global) hub
+        # and assigns it to the current thread's Sentry#get_current_hub
+        # it's essential for multi-thread servers (e.g. puma)
+        Sentry.clone_hub_to_current_thread unless Sentry.get_current_hub
+        # this call creates an isolated scope for every request
+        # it's essential for multi-process servers (e.g. unicorn)
+        Sentry.with_scope do |scope|
```

Comment:
because there are multi-thread/multi-process servers in the Ruby ecosystem, we need to support both of them.

for multi-thread servers, like `puma` or `webrick`, `Sentry.clone_hub_to_current_thread` sets up the a cloned hub in every new thread. so each thread's sentry scope, client...etc. won't interfere each other

for multi-process servers, like `unicorn`, the above line is redundant as every request will all run on the process' main thread. what it needs is an isolated scope for that particular request. so this is why we use `Sentry.with_scope` here.